rcl/src/rcl/logging_rosout.c

```diff
@@ -153,8 +154,10 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(
     return RCL_RET_ERROR;
   }
   if (rcutils_hash_map_key_exists(&__logger_map, &logger_name)) {
-    RCL_SET_ERROR_MSG("Logger already initialized for node.");
-    return RCL_RET_ALREADY_INIT;
+    // @todo: nburek enfornce node name unique
+    RCUTILS_LOG_WARN_NAMED("log_infrastructure", "Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from going out over the rosout topic.");
```

Comment:
Since `log_infrastructure` is not a term anywhere atm I would suggest to change it to: `rcl.logging_rosout`

And for the message instead of

> ... from going out over the rosout topic.

I would suggest

> ... from being published on the rosout topic.